app/ide-desktop/lib/dashboard/src/authentication/src/authentication/providers/session.tsx

```diff
@@ -25,18 +42,31 @@ const SessionContext = react.createContext<SessionContextType>(
 // =======================
 
 interface SessionProviderProps {
+  registerAuthEventListener: (callback: listen.ListenerCallback) => void;
   userSession: () => Promise<results.Option<cognito.UserSession>>;
   children: react.ReactNode;
 }
 
 // eslint-disable-next-line @typescript-eslint/naming-convention
 export const SessionProvider = (props: SessionProviderProps) => {
-  const { children, userSession } = props;
+  const { children, userSession, registerAuthEventListener } = props;
 
   /** Flag used to avoid rendering child components until we've fetched the user's session at least
    * once. Avoids flash of the login screen when the user is already logged in. */
   const [initialized, setInitialized] = react.useState(false);
 
+  /** State that, when incremented, forces a refresh of the user session. This is useful when a
+   * user has just logged in (so their cached credentials are out of date). Should be used via the
+   * `refreshSession` function. */
+  const [refresh, setRefresh] = react.useState(INITIAL_REFRESH_COUNT);
```

Comment:
consider using `useState({})` instead of a number. `({}) !== ({})` because objects use reference equality and every object literal is a different reference.

it will still require a comment but it should make the implementation simpler

Reply:
Even better, create a function like `useState(newState())` where `newState = () => {}` with good docs.